modules/database/src/Database.ts

```diff
@@ -54,6 +62,16 @@ export default class DatabaseModule extends ManagedModule {
   private adminRouter: AdminHandlers;
   private userRouter: DatabaseRoutes;
   private readonly _activeAdapter: DatabaseAdapter<MongooseSchema | SequelizeSchema>;
+  private readonly events: EventEmitter = new EventEmitter();
+  private _serviceHealthState: HealthCheckStatus = HealthCheckStatus.UNKNOWN;
+
+  private set serviceHealthState(state: HealthCheckStatus) {
```

Comment:
This could be simplified by the suggestion I added in a previous comment. It'll also remove the event emitter which is costly performance-wise in NodeJS.

Reply:
I moved this logit to `ConduitServiceModule`, but we can't just get rid of the emitter, unless we keep track of watch calls on the class and write back data on every state change directly inside `updateHealth()`.